libs/ojsonnet/Desugar_jsonnet_subst.ml

```diff
@@ -0,0 +1,505 @@
+(* Yoann Padioleau
```

Comment:
Again what's the difference with the original?
I was thinking we just need to have a Value_jsonnet_subst.ml and Eval_jsonnet_subst.ml.
We should try to keep Desugar and Core as-is, or if you need subst specific stuff to either
make the 2 original types polymorph or something so we can reuse them or clearly mark in the code
where the subst variant is different from the original code.